From ae719dd0ccaed80af4451fb7eca30d035f2f4392 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Mon, 25 Sep 2023 18:38:39 +0530
Subject: [PATCH] Sound: qcom: Add configs definition to build QCOM LPASS and
 PCM drivers

Define the LPASS and pcM platform driver,the LPASS,PCM and
PCM Test drivers configurations, and how to build them.

Change-Id: I71ca981acf8cbb372df7c2589372634c25095508
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
---
 sound/soc/qcom/Kconfig  | 15 +++++++++++++++
 sound/soc/qcom/Makefile |  1 +
 2 files changed, 16 insertions(+)

diff --git a/sound/soc/qcom/Kconfig b/sound/soc/qcom/Kconfig
index e7b00d1d9e99..d0c08298d865 100644
--- a/sound/soc/qcom/Kconfig
+++ b/sound/soc/qcom/Kconfig
@@ -221,4 +221,19 @@ config SND_SOC_SC7280
 	  SC7280 SoC-based systems.
 	  Say Y or M if you want to use audio device on this SoCs.
 
+config SND_SOC_IPQ
+	tristate "Soc Audio support for IPQ platforms"
+	depends on SND_SOC_QCOM && HAS_DMA
+	help
+	Say Y or M to if you want to add support for SoC audio on
+	Qualcomm Technologies IPQ based board.
+
+config SND_SOC_IPQ9574_LPASS_PCM_RAW
+	tristate "IPQ9574_LPASS_PCM"
+	depends on SND_SOC_IPQ
+
+config SND_SOC_IPQ_LPASS
+	tristate "IPQ_LPASS"
+	depends on SND_SOC_IPQ
+
 endif #SND_SOC_QCOM
diff --git a/sound/soc/qcom/Makefile b/sound/soc/qcom/Makefile
index 254350d9dc06..21adc85c1b03 100644
--- a/sound/soc/qcom/Makefile
+++ b/sound/soc/qcom/Makefile
@@ -40,6 +40,7 @@ obj-$(CONFIG_SND_SOC_SDM845) += snd-soc-sdm845.o
 obj-$(CONFIG_SND_SOC_SM8250) += snd-soc-sm8250.o
 obj-$(CONFIG_SND_SOC_QCOM_COMMON) += snd-soc-qcom-common.o
 obj-$(CONFIG_SND_SOC_QCOM_SDW) += snd-soc-qcom-sdw.o
+obj-$(CONFIG_SND_SOC_IPQ) += ipq/
 
 #DSP lib
 obj-$(CONFIG_SND_SOC_QDSP6) += qdsp6/
-- 
2.34.1

